data-manager/api/src/main/java/org/orbisgis/orbisdata/datamanager/api/dataset/ITable.java

```diff
@@ -48,37 +48,39 @@
  * @author Erwan Bocher (CNRS)
  * @author Sylvain PALOMINOS (UBS 2018-2019)
  */
-public interface ITable extends IDataSet {
+public interface ITable<T> extends IMatrix<T> {
 
     /**
      * Apply the given {@link Closure} to each row.
      *
      * @param closure {@link Closure} to apply to each row.
      */
-    void eachRow(Closure closure);
+    default void eachRow(Closure closure){
+        this.forEach(closure::call);
+    }
 
     /**
      * Get all column names from the underlying table. If there isn't any column, return an empty {@link Collection}.
      * If an error occurs on requesting the column, return null.
      *
      * @return A {@link Collection} containing the name of the column.
      */
-    Collection<String> getColumnNames();
+    Collection<String> getColumns();
 
     /**
      * Get all column information from the underlying table.
      *
      * @return A {@link Map} containing the information of the column.
      */
-    Map<String, String> getColumns();
+    Map<String, String> getColumnsTypes();
 
     /**
      * Get the type of the column from the underlying table.
      *
      * @param columnName set the name of the column
      * @return The type of the column.
      */
-    String getColumnsType(String columnName);
+    String getColumnType(String columnName);
```

Comment:
Will have impact on geoclimate and orbisanalysis

Reply:
Yes like other changes, but it is a nonsense to use `getColumnsType` when it return the type of only one column. The changes on geoclimate will come when the version used of OrbisData is updated.

Reply:
Just to keep in mind